lib/content_location_filter.js

```diff
@@ -29,6 +29,22 @@ function resolvedRedirectResponse(hyper, req, res) {
         // We don't pass the `origin` header further, so possible
         // redirect looping will be handled by normal redirect limiting in `preq`
     })
+    .then((res) => {
+        if (res.status >= 300) {
+            if (redirectNum > 10) {
+                throw new HTTPError({
+                    status: 504,
+                    body: {
+                        type: 'redirect loop',
```

Comment:
we set the 'type' ti an identifier of the http error, so it should be gateway_timeout for 504